test/integration/scheduler/rescheduling_test.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduler
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/scheduler"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	st "k8s.io/kubernetes/pkg/scheduler/testing"
+	testutils "k8s.io/kubernetes/test/integration/util"
+)
+
+var _ framework.PermitPlugin = &PermitPlugin{}
+var _ framework.EnqueueExtensions = &PermitPlugin{}
+
+type PermitPlugin struct {
+	name            string
+	statusCode      framework.Code
+	numPermitCalled int
+}
+
+func (pp *PermitPlugin) Name() string {
+	return pp.name
+}
+
+func (pp *PermitPlugin) Permit(ctx context.Context, state *framework.CycleState, p *v1.Pod, nodeName string) (*framework.Status, time.Duration) {
+	pp.numPermitCalled += 1
+
+	if pp.statusCode == framework.Error {
+		return framework.NewStatus(framework.Error, "failed to permit"), 0
+	}
+
+	if pp.statusCode == framework.Unschedulable {
+		if pp.numPermitCalled <= 1 {
+			return framework.NewStatus(framework.Unschedulable, "reject to permit"), 0
+		}
+	}
+
+	return nil, 0
+}
+
+func (pp *PermitPlugin) EventsToRegister() []framework.ClusterEventWithHint {
+	return []framework.ClusterEventWithHint{
+		{
+			Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add},
+			QueueingHintFn: func(pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+				return framework.QueueImmediately
+			},
+		},
+	}
+}
+
+func TestReScheduling(t *testing.T) {
+	testContext := testutils.InitTestAPIServer(t, "permit-plugin", nil)
+	tests := []struct {
+		name    string
+		plugins []framework.Plugin
+		action  func() error
+		// firstSchedulingError means for the first time in scheduling,
+		// pod encounters error, otherwise, pod is unschedulable.
+		// By default, pod is unschedulable for the first time.
+		firstSchedulingError bool
```

Comment:
this indicates an expected behavior not an "action", so lets prefix it with "want" like the other two below.

Reply:
Changed to wantFirstSchedulingError